commands/RunIfElse.java

```diff
@@ -1,66 +1,61 @@
 package org.usfirst.frc4904.standard.commands;
 
-
+import java.util.HashSet;
+import java.util.Set;
 import java.util.function.BooleanSupplier;
-import edu.wpi.first.wpilibj.command.Command;
-import edu.wpi.first.wpilibj.command.Subsystem;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Subsystem;
 
-public class RunIfElse extends Command {
+public class RunIfElse implements Command {
 	protected final Command ifCommand;
 	protected final Command elseCommand;
 	protected Command runningCommand;
 	protected final BooleanSupplier[] booleanSuppliers;
 	protected boolean hasRunOnce;
 
-	public RunIfElse(Command ifCommand, Command elseCommand, BooleanSupplier... booleanSuppliers) {
-		this("RunIf[" + ifCommand.getName() + "]Else[" + elseCommand.getName() + "]", ifCommand, elseCommand, booleanSuppliers);
-	}
+	protected final Set<Subsystem> unionRequirements;
 
-	protected RunIfElse(String name, Command ifCommand, Command elseCommand, BooleanSupplier... booleanSuppliers) {
-		super(name);
+	protected RunIfElse(Command ifCommand, Command elseCommand, BooleanSupplier... booleanSuppliers) {
 		this.ifCommand = ifCommand;
 		this.elseCommand = elseCommand;
 		this.booleanSuppliers = booleanSuppliers;
-	}
 
-	@Override
-	public boolean doesRequire(Subsystem subsystem) {
-		return ifCommand.doesRequire(subsystem) || elseCommand.doesRequire(subsystem);
+		unionRequirements = new HashSet<Subsystem>(ifCommand.getRequirements());
+		unionRequirements.addAll(elseCommand.getRequirements());
```

Comment:
`addRequirements(ifCommand.getRequirements());
addRequirements(elseCommand.getRequirements());`

Reply:
I don't understand, should I add those lines or change the noted lines to those?